arviz/stats/stats.py

```diff
@@ -246,7 +246,7 @@ def compare(
         try:
             # Here is where the IC function is actually computed -- the rest of this
             # function is argument processing and return value formatting
-            ics = ics.append([ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)])
+            ics = pd.concat([ics, ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)])
```

Comment:
I think you can replace this whole piece with something like
```
ics = pd.concat({key: ic_func(val, pointwise=True, scale=scale, var_name=var_name) for key, val in dataset_dict.items()}, axis=1).T
```
as part of the reason for deprecating `append` is that it's slow

Reply:
What about the exception?

Reply:
you could append these internal results to some other `dict` and then call `concat` at the end

or make a wrapped around `ic_func` which raises the exception if it fails